src/olympia/scanners/tasks.py

```diff
@@ -325,9 +325,20 @@ def _run_yara_query_rule_on_version(version, rule):
     """
     file_ = version.all_files[0]
     scanner_result = ScannerQueryResult(version=version, scanner=YARA)
-    _run_yara_for_path(
-        scanner_result, file_.current_file_path,
-        definition=rule.definition)
+    try:
+        _run_yara_for_path(
+            scanner_result, file_.current_file_path,
+            definition=rule.definition)
+    except FileNotFoundError:
+        # Fallback in case the file was disabled/re-enabled and not yet moved,
+        # we try the other possible path. This shouldn't happen too often.
+        tried_path = file_.current_file_path
+        fallback_path = (
+            file_.file_path if tried_path == file_.guarded_file_path
+            else file_.guarded_file_path
+        )
+        _run_yara_for_path(
+            scanner_result, fallback_path, definition=rule.definition)
```

Comment:
Is it possible that this `fallback_path` will also yield a `FileNotFoundError`?

Reply:
Yes: there could be a race condition, or the file could simply be missing for any number of reasons (since we're running searches on potentially old/buggy versions we have in the db). In that case it would simply raise the exception as normal, the parent task catches all exceptions, logs them and continues.